src/browser/BrowserAccessControlDialog.cpp

```diff
@@ -25,42 +25,81 @@ BrowserAccessControlDialog::BrowserAccessControlDialog(QWidget* parent)
     : QDialog(parent)
     , m_ui(new Ui::BrowserAccessControlDialog())
 {
-    this->setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
+    setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
 
     m_ui->setupUi(this);
-    connect(m_ui->allowButton, SIGNAL(clicked()), this, SLOT(accept()));
-    connect(m_ui->denyButton, SIGNAL(clicked()), this, SLOT(reject()));
+
+    connect(m_ui->allowButton, SIGNAL(clicked()), SLOT(accept()));
+    connect(m_ui->cancelButton, SIGNAL(clicked()), SLOT(reject()));
 }
 
 BrowserAccessControlDialog::~BrowserAccessControlDialog()
 {
 }
 
-void BrowserAccessControlDialog::setUrl(const QString& url)
+void BrowserAccessControlDialog::setItems(const QList<Entry*>& items, const QString& hostname, bool httpAuth)
 {
-    m_ui->label->setText(QString(tr("%1 has requested access to passwords for the following item(s).\n"
-                                    "Please select whether you want to allow access."))
-                             .arg(QUrl(url).host()));
-}
+    m_ui->siteLabel->setText(m_ui->siteLabel->text().arg(hostname));
 
-void BrowserAccessControlDialog::setItems(const QList<Entry*>& items)
-{
-    for (Entry* entry : items) {
-        m_ui->itemsList->addItem(entry->title() + " - " + entry->username());
+    m_ui->rememberDecisionCheckBox->setVisible(!httpAuth);
+    m_ui->rememberDecisionCheckBox->setChecked(false);
+
+    m_ui->itemsTable->setRowCount(items.count());
+    m_ui->itemsTable->setColumnCount(2);
+
+    int row = 0;
+    for (const auto& entry : items) {
+        auto item = new QTableWidgetItem();
+        item->setText(entry->title() + " - " + entry->username());
+        item->setData(Qt::UserRole, row);
+        item->setCheckState(Qt::Checked);
+        item->setFlags(item->flags() | Qt::ItemIsUserCheckable);
+        m_ui->itemsTable->setItem(row, 0, item);
+
+        auto disableButton = new QPushButton(tr("Disable for this site"));
+        connect(disableButton, &QAbstractButton::pressed, [=] {
```

Comment:
You only need to capture `item` and `this` here.

Reply:
Lambda expressions only capture the variables that are actually used. I should have used `&` instead, fixed.